ui/raidboss/data/07-dt/raid/r8s.ts

```diff
@@ -1,14 +1,824 @@
+        const actors = (await callOverlayHandler({
+          call: 'getCombatants',
+          ids: [parseInt(matches.sourceId, 16)],
+        })).combatants;
+        const actor = actors[0];
+        if (actors.length !== 1 || actor === undefined) {
+          console.error(
+            `R8S Beckon Moonlight Quadrants: Wrong actor count ${actors.length}`,
+          );
+          return;
+        }
+
+        const dirNum = Directions.xyTo8DirNum(actor.PosX, actor.PosY, centerX, centerY);
+        // Moonbeam's Bite (A3C2 Left / A3C3 Right) half-room cleaves
+        // Defining the cleaved side
+        if (matches.id === 'A3E0') {
+          const counterclock = dirNum === 0 ? 6 : dirNum - 2;
+          data.moonbeamBites.push(counterclock);
+        }
+        if (matches.id === 'A3E1') {
+          const clockwise = (dirNum + 2) % 8;
+          data.moonbeamBites.push(clockwise);
+        }
+      },
+      infoText: (data, _matches, output) => {
+        if (data.moonbeamBites.length === 1 || data.moonbeamBites.length === 3)
+          return;
+
+        const quadrants = [1, 3, 5, 7];
+        // When there are multiple safe spots, output cardinal
+        const intersToCard = (dirNum1: number, dirNum2: number) => {
+          // Northeast and Northwest
+          if (dirNum1 === 1 && dirNum2 === 7 || dirNum2 === 7 && dirNum1 === 1)
+            return 0;
+          // Northeast and Southeast
+          if (dirNum1 === 1 && dirNum2 === 3 || dirNum1 === 3 && dirNum2 === 1)
+            return 2;
+          // Southeast and Southwest
+          if (dirNum1 === 3 && dirNum2 === 5 || dirNum1 === 5 && dirNum2 === 3)
+            return 4;
+          // Southwest and Northwest
+          if (dirNum1 === 5 && dirNum2 === 7 || dirNum1 === 7 && dirNum2 === 5)
+            return 6;
+        };
+
+        const moonbeam1 = data.moonbeamBites[0] ?? -1;
+        const moonbeam2 = data.moonbeamBites[1] ?? -1;
+        let safeQuads1 = quadrants.filter((quadrant) => {
+          return quadrant !== moonbeam1 + 1;
+        });
+        safeQuads1 = safeQuads1.filter((quadrant) => {
+          return quadrant !== moonbeam1 - 1;
+        });
+        safeQuads1 = safeQuads1.filter((quadrant) => {
+          return quadrant !== moonbeam2 + 1;
+        });
+        safeQuads1 = safeQuads1.filter((quadrant) => {
+          return quadrant !== moonbeam2 - 1;
+        });
+
+        // Early output for first two
+        if (data.moonbeamBites.length === 2) {
+          if (safeQuads1.length === 2) {
+            if (safeQuads1[0] === undefined || safeQuads1[1] === undefined) {
+              console.error(
+                `R8S Beckon Moonlight Quadrants: Early safeQuad missing.`,
+              );
+              return;
+            }
+            const dirNum = intersToCard(safeQuads1[0], safeQuads1[1]);
+            const half = output[Directions.outputFrom8DirNum(dirNum ?? -1)]!();
+            return output.safeHalf!({ half: half });
+          }
```

Comment:
Is this code reachable? Since the adds always spawn in a CW / CCW pattern, it should not be possible for the same half of the arena to be safe for two slashes in a row. This applies to anything that uses `intersToCard`.

Reply:
It gets reached, which I think means there may be an issue with the code.

Reply:
It may be that there needs to be some more delay in certain positions, I have only seen it call W first.

Reply:
The error seems to happen because it received an empty moonbeam2 value. I'll dig more into it later.

Reply:
Found out that the same check for 0 in the position capture needs to happen again on the filter since the math for the safe side can result in a 0 as well.